qiskit/primitives/base/base_sampler.py

```diff
@@ -278,7 +278,7 @@ class BaseSamplerV2:
 
     @abstractmethod
     def run(
-        self, pubs: Iterable[SamplerPubLike], shots: int | None = None
+        self, pubs: Iterable[SamplerPubLike], *, shots: int | None = None
```

Comment:
Thanks. @chriseclectic also added this to the BaseEstimatorV2 PR which was merged.